### PR TITLE
[WIP] Add support sf4

### PR DESCRIPTION
**Pull request by stood**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ env:
     matrix:
         - SYMFONY_VERSION=2
         - SYMFONY_VERSION=3
+        - SYMFONY_VERSION=4
```
Comment (sanpii) [inline .travis.yml]: This is not magic, you should modify something bellow.

**Comment by mvrhov** [comment]
My PR opened 8 moths ago should be merged first, and then this one rebased on top of it.

**Comment by EvKoh** [comment]
+1 for this PR !!!

**Comment by stood** [comment]
@chanmix51  can you tag CLI plz for support 4.0 ?

**Comment by sanpii** [comment]
Ok, I think I have understood the problem with php5.4 and symfony 3/4 build.

PHP 5.4 force symfony to keep in 2.8 version: https://travis-ci.org/pomm-project/pomm-bundle/jobs/323213244#L612

We need a better way to dertermine if we should launch `@upper2lts` tagged tests.

```bash
current_version=$(composer show --format=json | jq '.installed[] | select(.name == "symfony/symfony") | .version' | sed 's/^"v\([^"]*\)"/\1/')
sf2 = $(dpkg --compare-versions "$current_version" "lt" "3"; echo $?)
test $sf2 == 0 && ./bin/behat --tags '~@upper2lts' ||./bin/behat
```

File: .travis.yml [hunk 2]
```diff
@@ -36,21 +37,25 @@ before_install:
 
     - php -S localhost:8080 -t tests/web &> /dev/null &
     - ln -fs parameters.yml.dist tests/app/config/parameters.yml
+    - test "$SYMFONY_VERSION" -eq 2 && composer require --no-update symfony/lts:^2||true
+    - test "$SYMFONY_VERSION" -eq 3 && composer require --no-update symfony/lts:^3||true
+    - cd tests/
+    - test "$SYMFONY_VERSION" -eq 2 && composer require --no-update symfony/lts:^2||true
+    - test "$SYMFONY_VERSION" -eq 3 && composer require --no-update symfony/lts:^3||true
```
Comment (sanpii) [inline .travis.yml]: These lines can probably be merged like:
```
composer require --no-update "symfony/lts:^$SYMFONY_VERSION"
```

File: .travis.yml [hunk 3]
```diff
@@ -36,21 +37,25 @@ before_install:
 
     - php -S localhost:8080 -t tests/web &> /dev/null &
     - ln -fs parameters.yml.dist tests/app/config/parameters.yml
+    - test "$SYMFONY_VERSION" -eq 2 && composer require --no-update symfony/lts:^2||true
+    - test "$SYMFONY_VERSION" -eq 3 && composer require --no-update symfony/lts:^3||true
```
Comment (sanpii) [inline .travis.yml]: These lines can probably be merged like:
```
composer require --no-update "symfony/lts:^$SYMFONY_VERSION"
```

File: .travis.yml [hunk 4]
```diff
@@ -36,21 +37,25 @@ before_install:
...
-    - composer install
-    - test "$SYMFONY_VERSION" -eq 3 || composer update --prefer-lowest
+    - composer update
     - rm -rf vendor/pomm-project/pomm-bundle
     - ln -s ../../../ vendor/pomm-project/pomm-bundle
 
 script:
     - ../vendor/bin/phpcs --standard=psr2 --runtime-set ignore_warnings_on_exit true --report=summary ../sources
-    - ./bin/behat
+    - test "$SYMFONY_VERSION" -eq 2 && ./bin/behat --tags '~@upper2lts' ||./bin/behat
```
Comment (sanpii) [inline .travis.yml]: A probably more readable version:
```
./bin/behat --tags "~@upper${SYMFONY_VERSION}lts"
```

**Pull request merged**